proposed/links.md

```diff
@@ -162,14 +184,92 @@ interface LinkInterface
 }
 ~~~
 
-#### 3.2.1 `Psr\Http\Link\LinkCollectionInterface`
+### 3.2 `Psr\Http\Link\EvolvableLinkInterface`
 
 ~~~php
 <?php
+
 namespace Psr\Http\Link;
 
 /**
- *
+ * An evolvable link value object.
+ */
+interface EvolvableLinkInterface extends LinkInterface
+{
+    /**
+     * Returns an instance with the specified href.
+     *
+     * @param string $href
+     *   The href value to include.  It must be one of:
+     *     - An absolute URI, as defined by RFC 5988.
+     *     - A relative URI, as defined by RFC 5988. The base of the relative link
+     *       is assumed to be known based on context by the client.
+     *     - A URI template as defined by RFC 6570.
+     *     - An object implementing __toString() that produces one of the above
+     *       values.
+     *
+     * An implementing library SHOULD evaluate a passed object to a string
+     * immediately rather than waiting for it to be returned later.
+     *
+     * @return static
+     */
+    public function withHref($href);
+
+    /**
+     * Returns an instance with a specified templated value set.
+     *
+     * @param bool $templated
+     *   True if the link object should be templated, False otherwise.
+     * @return static
+     */
+    public function withTemplated($templated);
+
+    /**
+     * Returns an instance with the specified relationship included.
+     *
+     * If the specified rel is already present, this methid MUST return
+     * normally without errors, but without adding the rel a second time.
+     *
+     * @param string $rel
+     *   The relationship value to add.
+     * @return static
+     */
+    public function withRel($rel);
```

Comment:
The main interface has getRel(), so it makes sense to be consistent with that.  If we were to change it, we'd need to change both.

We went with "rel" primarily because that's what implementations tend to use, if anything.  Both HTML and HTTP Link headers use "rel", not "relationship".  The wisdom of that decision is debatable, but it is what it is, and since that property maps directly to those representations it makes sense to use the same name throughout.